### DISCUSSION TITLE
Reaction is not getting fired correctly

### DISCUSSION THREAD

**Discussion by smriti-chawla** [mobxjs/mobx#2561] [Q&A] [answered]
```
  matchEventTitles = reaction(
    () => this.activeImpact.events.map(event => event),
    (eventDetails): void => {
          console.log('doing some function work here', eventDetails)
    },
  );
```
activeImpact is an observable like below

```
@observable activeImpact: Impact = {
    events: [],
  };
```

the reaction is getting fired whenever there is a change in the length of **this.activeImpact.events**.
However, I want to fire this whenever there is any change in this.activeImpact.events whether it's a specific property to it.
for eg if I do something like 
  ```
matchEventTitles = reaction(
    () => this.activeImpact.events.map(event => ({
        id: event.id,
        name: event.name,
      })),
    (eventDetails): void => {
      console.log('doing some function work here', eventDetails)
      
    },
  );
```

then in that case the reaction is working for change in property id and name .

however, there are so many properties in this I cannot write then like this to observe the change

**Comment by danielkcz** [comment]
Try `autorun` instead.

**Reply by smriti-chawla** [reply]
autorun is just getting fired once when the page loads. 

Why is it not working with reaction? since  () => this.activeImpact.events.map(event => event), this would any change in the array of object I believe

**Reply by danielkcz** [reply]
Best if you can setup reproduction with eg. CodeSandbox.

**Comment by mweststrate** [comment, marked as answer]
See https://mobx.js.org/understanding-reactivity.html, mobx only tracks
what you are actually reading; in your example you are reading the keys and
values of the map, but not any attributes of the events itself, so those
won't be tracked.

**Reply by danielkcz** [reply]
I still seem to have a problem grasping these basic as well 😅

**Reply by smriti-chawla** [reply]
```
 matchEventTitles = reaction(
    () => this.activeImpact.events.map(event =>{ return {...event}}),
    (eventDetails): void => {
          console.log('doing some function work here', eventDetails)
    },
  );

```
this is working for it. Added restructuring so every time a new instance is returned

**Reply by danielkcz** [reply]
@smriti-chawla I believe you could also do `mobx.toJS(this.activeImpact.events)` with the same effect unless you want to do some modifications in that map.

**Reply by smriti-chawla** [reply]
@FredyC correct. Thank you
